src/vs/workbench/browser/parts/editor/media/tabstitlecontrol.css

```diff
@@ -108,20 +113,12 @@
 	display: none;
 }
 
-.monaco-workbench .part.editor > .content .editor-group-container > .title .tabs-container > .tab:first-of-type {
-	padding-left: 5px !important;
-}
-
-.monaco-workbench .part.editor > .content .editor-group-container > .title .tabs-container > .tab:last-of-type {
-	padding-right: 5px !important;
+.monaco-workbench .part.editor > .content .editor-group-container.active > .title .tabs-container > .tab:hover > .tab-actions .action-label {
+	border-radius: 50%;
 }
 
 .monaco-workbench .part.editor > .content .editor-group-container > .title > .tabs-and-actions-container.wrapping .tabs-container > .tab:last-child {
-	margin-right: var(--last-tab-margin-right); /* when tabs wrap, we need a margin away from the absolute positioned editor actions */
-}
-
-.monaco-workbench .part.editor > .content .editor-group-container > .title > .tabs-and-actions-container.wrapping .tabs-container > .tab.last-in-row:not(:last-child) {
-	border-right: 0 !important; /* ensure no border for every last tab in a row except last row (#115046) */
```

Comment:
Custom CSS does not substitute the CSS files themselves. If we remove the selector in this scenario, it will not be removed from vscode. Our objective should be to override its properties in order to disable it. For instance, we can use display: none or an equivalent method.

Reply:
Yes, see line 99:

```diff
+ border: none !important; /* ensure no border for every tab (#3) */
```

It's inherited by all tab states so no need to specify on a case-by-case basis

Reply:
> Yes, see line 99:
> 
> ```diff
> + border: none !important; /* ensure no border for every tab (#3) */
> ```
> 
> It's inherited by all tab states so no need to specify on a case-by-case basis

To clarify: The content is actually sourced from vscode itself, and our goal is to maintain a mirrored view from upstream - that's why we choose to preserve things as they are and make our modifications as an "overlay." While it may seem like a good idea to remove it, since we function as an overlay, we prefer to keep it there and simply disable or comment it out, so that it is still be present in the source, even if it doesn't have any actual impact ~ 

One of the reasons for this approach is to ensure convenience in locating and understanding the default values of selectors. It is particularly important for us to stay updated with any upstream changes that could potentially affect the theme and, by keeping things as they are, we can effectively monitor and address any modifications that might impact it in the future

Reply:
Ok, makes sense :+1: 
are you able to modify the PR?

Reply:
Certainly, I will do once I'm available tomorrow :)

Reply:
I also [started on Find (Ctrl-f)](https://github.com/Acumane/libadwaita-vscode-theme/commit/d114522157d54a5adc5a8f7deabdb19598691d23). Something like this, I'm thinking:

![Screenshot from 2023-05-13 23-11-48](https://github.com/wroyca/libadwaita-vscode-theme/assets/94324138/f7c447f1-2093-4d34-9315-ed129ae64709)

Reply:
That sounds wonderful! 

Just to inform you, I have merged your changes. However, since you committed your [Find (Ctrl-f)](https://github.com/Acumane/libadwaita-vscode-theme/commit/d114522157d54a5adc5a8f7deabdb19598691d23) work on the same branch as this pull request, I had to selectively choose and apply the commit. As a result, I will be closing this merge request, but rest assured that your previous modification has been successfully committed: https://github.com/wroyca/libadwaita-vscode-theme/commit/ca71589714c88009c385e0029ff6ec8362f06bb0